dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-prometheus/src/main/java/org/apache/dolphinscheduler/plugin/alert/prometheus/AlertManagerSender.java

```diff
@@ -0,0 +1,169 @@
+
+    private String url;
+    private String generatorURL;
+    private String annotations;
+
+    public AlertManagerSender(Map<String, String> config) {
+        url = config.get(AlertManagerConstants.NAME_ALERT_MANAGER_URL);
+        generatorURL = config.get(AlertManagerConstants.NAME_GENERATOR_URL);
+        annotations = config.get(AlertManagerConstants.NAME_ALERT_MANAGER_ANNOTATIONS);
+    }
+
+    public AlertResult sendMessage(AlertData alertData) {
+        AlertResult alertResult;
+        try {
+            String resp = sendMsg(alertData);
+            return checkSendAlertManageMsgResult(resp);
+        } catch (Exception e) {
+            log.info("send prometheus alert manager alert msg  exception: {}", e.getMessage());
+            alertResult = new AlertResult();
+            alertResult.setStatus("false");
+            alertResult.setMessage("send prometheus alert manager alert fail.");
+        }
+        return alertResult;
+    }
+
+    private String sendMsg(AlertData alertData) throws IOException {
+        String v2Path = String.format("%s%s", this.url, AlertManagerConstants.ALERT_V2_API_PATH);
+        String msg = generateContentJson(alertData);
+        HttpPost httpPost = constructHttpPost(v2Path, msg);
+        CloseableHttpClient httpClient = getDefaultClient();
+
+        try {
+            CloseableHttpResponse response = httpClient.execute(httpPost);
+            String resp;
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode == HttpStatus.SC_OK) {
+                resp = AlertManagerConstants.ALERT_SUCCESS;
+                log.info("Prometheus alert manager send alert succeed, title: {} ,content: {}", alertData.getTitle(),
+                        alertData.getContent());
+                return resp;
+            }
+
+            log.error("send prometheus alert manager message error, return http status code: {} ", statusCode);
+            try {
+                HttpEntity entity = response.getEntity();
+                resp = EntityUtils.toString(entity, "utf-8");
+                EntityUtils.consume(entity);
+            } finally {
+                response.close();
+            }
+            log.info("Prometheus alert manager send alert failed, title: {} ,content: {}, resp: {}",
+                    alertData.getTitle(),
+                    alertData.getContent(), resp);
+
+            return resp;
+        } finally {
+            httpClient.close();
+        }
+    }
+
+    public AlertResult checkSendAlertManageMsgResult(String resp) {
+        AlertResult alertResult = new AlertResult();
+        alertResult.setStatus("false");
+
+        if (Objects.equals(resp, AlertManagerConstants.ALERT_SUCCESS)) {
+            alertResult.setStatus("true");
+            alertResult.setMessage("prometheus alert manager send success");
+            return alertResult;
+        }
+
+        alertResult.setMessage("prometheus alert manager send fail");
```

Comment:
Please set the code and response in `alertResult.setMessage`. It's easier to find the error reason.

Reply:
Ok.